erpc_c/setup/erpc_setup.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2014-2016, Freescale Semiconductor, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * o Redistributions of source code must retain the above copyright notice, this list
+ *   of conditions and the following disclaimer.
+ *
+ * o Redistributions in binary form must reproduce the above copyright notice, this
+ *   list of conditions and the following disclaimer in the documentation and/or
+ *   other materials provided with the distribution.
+ *
+ * o Neither the name of Freescale Semiconductor, Inc. nor the names of its
+ *   contributors may be used to endorse or promote products derived from this
+ *   software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _EMBEDDED_RPC_SETUP_H_
+#define _EMBEDDED_RPC_SETUP_H_
+#include <new>
+#include <assert.h>
+
+#if !(__embedded_cplusplus)
+using namespace std;
+#endif
+
+using namespace erpc;
+
+////////////////////////////////////////////////////////////////////////////////
+// CLASS
+////////////////////////////////////////////////////////////////////////////////
+class BasicMessageBufferFactory : public MessageBufferFactory
+{
+public:
+    virtual MessageBuffer create()
+    {
+        uint8_t *buf = new (nothrow) uint8_t[ERPC_DEFAULT_BUFFER_SIZE];
+        return MessageBuffer(buf, ERPC_DEFAULT_BUFFER_SIZE);
+    }
+
+    virtual void dispose(MessageBuffer *buf)
+    {
+        assert(buf);
+        if (*buf)
+        {
+            delete[] buf->get();
+        }
+    }
+};
+
```

Comment:
1.  right with EMBEDDED_RPC_SETUP_H end
2.  as the include file will not be processed twice as we add the `#ifndef `in each header file, I think it is not a must. Besides, why not use the `#pragma` once instead of `#ifndef ...  #endif` pairs for such conditions?
3. move the static variable from server and client to a common place will have problem when you application have both server and client functions. e.g. in my test application, I use a run time command to switch between client and server, which make the static variable remains for server/client is better.